keylime/tpm/tpm_abstract.py

```diff
@@ -140,67 +151,82 @@ def _set_tpm_metadata(self, key, value):
             self.__write_tpm_data()
 
     @abstractmethod
-    def tpm_init(self, self_activate=False, config_pw=None):
+    def tpm_init(self, self_activate: bool = False, config_pw: Optional[str] = None):
         pass
 
     # tpm_quote
     @abstractmethod
-    def create_quote(self, nonce, data=None, pcrmask=EMPTYMASK, hash_alg=None, compress=False):
+    def create_quote(
+        self,
+        nonce: str,
+        data: Optional[bytes] = None,
+        pcrmask: str = EMPTYMASK,
+        hash_alg: Optional[str] = None,
+        compress: bool = False,
+    ):
         pass
 
     @abstractmethod
     def check_quote(
         self,
-        agentAttestState,
-        nonce,
-        data,
-        quote,
-        aikTpmFromRegistrar,
-        tpm_policy=None,
-        ima_measurement_list=None,
-        allowlist=None,
-        hash_alg=None,
-        ima_keyrings=None,
-        mb_measurement_list=None,
-        mb_refstate=None,
-        compressed=False,
+        agentAttestState: AgentAttestState,
+        nonce: str,
+        data: str,
+        quote: str,
+        aikTpmFromRegistrar: str,
+        tpm_policy: Optional[Union[str, Dict]] = None,
```

Comment:
The `tpm_policy` is a candidate for  cleanup later on. It should just be a `Dict[str, str]` or `Dict[int, str]`...